packages/foundation/src/next/composed.tsx

```diff
@@ -42,8 +43,96 @@ export function composed<
+ * and styled before finally being passed to view.
+ *
+ * State components should contain all stateful behavior and should not generate any JSX, but rather simply call the view prop.
+ * Views should simply be stateless pure functions that receive all props needed for rendering their output.
+ * State component is optional. If state is not provided, created component is essentially a functional stateless component.
+ *
+ * @param baseComponent - base component to recompose
+ * @param options - component Component options. See IComponentOptions for more detail.
+ */
+export function composed<
+  TComponentProps extends ValidProps & ISlottableProps<TComponentSlots>,
+  TTokens,
+  TStyleSet extends IStyleSet<TStyleSet>,
+  TViewProps extends TComponentProps = TComponentProps,
+  TComponentSlots = {},
+  TStatics = {}
+>(
+  baseComponent: React.FunctionComponent,
+  options: IRecompositionComponentOptions<TComponentProps, TTokens, TStyleSet, TViewProps, TComponentSlots, TStatics>
+): React.FunctionComponent<TComponentProps> & TStatics;
+
+export function composed<
+  TComponentProps extends ValidProps & ISlottableProps<TComponentSlots>,
+  TTokens,
+  TStyleSet extends IStyleSet<TStyleSet>,
+  TViewProps extends TComponentProps = TComponentProps,
+  TComponentSlots = {},
+  TStatics = {}
+>(
+  baseComponentOrOptions:
+    | React.FunctionComponent
+    | IComponentOptions<TComponentProps, TTokens, TStyleSet, TViewProps, TComponentSlots, TStatics> = {},
+  recompositionOptions?: IRecompositionComponentOptions<TComponentProps, TTokens, TStyleSet, TViewProps, TComponentSlots, TStatics>
 ): React.FunctionComponent<TComponentProps> & TStatics {
+  const baseComponentOptions = (baseComponentOrOptions as ISlotCreator<
+    TComponentProps,
+    any,
+    TTokens,
+    TStyleSet,
+    TViewProps,
+    TComponentSlots,
+    TStatics
+  >).__options;
+
+  // Check if we are composing or recomposing.
+  let options: IComponentOptions<TComponentProps, TTokens, TStyleSet, TViewProps, TComponentSlots, TStatics>;
+  if (typeof baseComponentOrOptions === 'function' && baseComponentOptions) {
+    let slots: ISlotComponent<TComponentProps, TComponentSlots>;
+    const baseComponentSlots = baseComponentOptions.slots ? baseComponentOptions.slots : ({} as ISlotDefinition<Required<TComponentSlots>>);
+    const recompositionSlots = recompositionOptions ? (recompositionOptions.slots ? recompositionOptions.slots : {}) : {};
+
+    if (typeof baseComponentSlots === 'function') {
+      if (typeof recompositionSlots === 'function') {
+        slots = props => {
+          return { ...baseComponentSlots(props), ...recompositionSlots(props) };
+        };
+      } else {
+        slots = props => {
+          return { ...baseComponentSlots(props), ...recompositionSlots };
+        };
+      }
+    } else {
+      if (typeof recompositionSlots === 'function') {
```

Comment:
really having a hard time following this. Why the need to resolve functional slot collection?

Reply:
If slots can be functions then the resolution for `base` + `options` would need to take this into account, wouldn't it?

Reply:
nevermind!